ocp_resources/resource.py

```diff
@@ -1059,27 +1060,40 @@ def wait_for_condition(self, condition: str, status: str, timeout: int = 300) ->
                     if cond["type"] == condition and cond["status"] == status:
                         return
 
-    def api_request(self, method: str, action: str, url: str, **params: Any) -> dict[str, Any]:
+    def api_request(
+        self, method: str, action: str, url: str, retry_params: dict[str, int] | None = None, **params: Any
+    ) -> dict[str, Any]:
         """
         Handle API requests to resource.
 
         Args:
             method (str): Request method (GET/PUT etc.).
             action (str): Action to perform (stop/start/guestosinfo etc.).
             url (str): URL of resource.
+            retry_params (dict): dict of timeout and sleep_time values for retrying the api request call
 
         Returns:
            data(dict): response data
 
         """
         client: DynamicClient = self.client
-        response = client.client.request(
-            method=method,
-            url=f"{url}/{action}",
-            headers=client.client.configuration.api_key,
-            **params,
-        )
-
+        if retry_params:
+            response = self.retry_cluster_exceptions(
+                func=client.client.request,
+                method=method,
+                url=f"{url}/{action}",
+                headers=client.client.configuration.api_key,
```

Comment:
Please save `method`, `url` and `headers` in dict and reuse them in both calls with `**`